### PR TITLE
Xentop parser

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: xentop/xentop.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (al1img) [inline xentop/xentop.go:1]: Move it under utils folder

File: xentop/xentop.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+// SystemInfo domain info.
+type SystemInfo struct {
+	Name               string  `field:"NAME"`
+	State              string  `field:"STATE"`
+	CpuTime            int64   `field:"CPU(sec)"` //nolint:stylecheck
```
Comment (al1img) [inline xentop/xentop.go:39]: CPUTime, CPUFraction looks ok.

File: xentop/xentop.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+			return systemInfo, aoserrors.New("unexpected field type in SystemInfo struct")
+		}
+	}
+
+	return systemInfo, nil
+}
+
+func newExecShellCommander(name string, arg ...string) ShellCommand {
+	execCmd := exec.Command(name, arg...)
+	return execShellCommand{Cmd: execCmd}
```
Comment (al1img) [inline xentop/xentop.go:192]:
```go
return execShellCommand{Cmd: exec.Command(name, arg...)}
```

File: xentop/xentop.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+		if len(info) != len(header) {
+			return nil, aoserrors.New("unexpected xentop data")
+		}
+
+		systemInfo, err := fillDomainInfo(header, info)
+		if err != nil {
+			return nil, err
+		}
+
+		// Ignore parse system information if xentop show extremely high CPU time and CPU usage
```
Comment (al1img) [inline xentop/xentop.go:109]: Why we should ignore it? It is xentop problem or some system problem.

File: xentop/xentop.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+	header := strings.Fields(xentopLines[0])
+	if len(header) < 1 || header[0] != "NAME" {
+		return nil, aoserrors.New("unexpected xentop header")
+	}
+
+	systemInfos := make(map[string]SystemInfo)
+
+	for _, line := range xentopLines[1:] {
+		line = strings.ReplaceAll(line, "no limit", "no-limit")
```
Comment (al1img) [inline xentop/xentop.go:97]: Need comment here: why we should replace it.

File: xentop/xentop.go (hunk 1 above)
Comment (MykolaSolyanko) [reply xentop/xentop.go:1]: Done

File: xentop/xentop.go (hunk 4 above)
Comment (MykolaSolyanko) [reply xentop/xentop.go:109]: Agree. Remove it

File: xentop/xentop.go (hunk 3 above)
Comment (MykolaSolyanko) [reply xentop/xentop.go:192]: Done

File: xentop/xentop.go (hunk 5 above)
Comment (MykolaSolyanko) [reply xentop/xentop.go:97]: Done

File: utils/xentop/xentop.go [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+// SystemInfo domain info.
+type SystemInfo struct {
+	Name               string  `field:"NAME"`
+	State              string  `field:"STATE"`
+	CPUTime            int64   `field:"CPU(sec)"` //nolint:stylecheck
```
Comment (al1img) [inline utils/xentop/xentop.go:37]: Still //nolint required?

Comment (MykolaSolyanko) [reply utils/xentop/xentop.go]: Forgot to remove it

**Review by al1img — APPROVED** [review]

**Review by Eabramov84 — APPROVED** [review]

**Pull request merged**